### ISSUE TITLE
[Bug] Properties not recognized by eslint-plugin

### ISSUE THREAD

**Issue by olivierpascal** [facebook/stylex#135]
**The problem**

The `outlineOffset` property is not recognized by the eslint-plugin.

**How to reproduce**

With the following code:

```ts
import * as stylex from '@stylexjs/stylex';

export const styles = stylex.create({
  container: {
    outlineOffset: '2px', // eslint(@stylexjs/valid-styles) error here
  },
});
```

I have the following eslint error from `@stylex/valid-styles`:

```
outlineOffset value must be one of:
'outlineOffset' is not supported yet. Please use 'outline' instead
```

https://github.com/facebook/stylex/blob/f968076548895dd1adc9736f1bfc371057beb94e/packages/eslint-plugin/src/stylex-valid-styles.js#L1948-L1950

But `outline` syntax does not allow an offset:

```
outline = 
  <'outline-color'>  ||
  <'outline-style'>  ||
  <'outline-width'>
```
Cf. https://developer.mozilla.org/en-US/docs/Web/CSS/outline

**Comment by olivierpascal** [comment]
Same for `transition`:

```ts
export const styles = stylex.create({
  container: {
    transition: 'box-shadow 150ms cubic-bezier(0.4, 0, 0.2, 1)', // eslint(@stylexjs/valid-styles) error here
  },
```

```
This is not a key that is allowed by stylex eslint(@stylexjs/valid-styles)
```

**Comment by nmn** [comment]
NOTE: A few properties are intentionally not allowed by the ESLint plugin even though they will work correctly in the Babel plugin:
- `background`
- `transition`
- `animation`
- `border`, `border[Direction]`.
- `flex`

This decision was made for a few reasons:
1. To encourage better styling practices. We've noticed that a lot of developers don't understand what the various parts of some of these properties are.
2. To enable smaller CSS. `transitions`, `animations`, and `background` are all properties where many combinations of the same values may be used.
3. To align a bit with React Native for our future RN implementation.

We are going to be discussing and reconsidering this decision.

---

Any other missing property is an unknown bug so please report!

**Comment by olivierpascal** [comment]
Well noted, thanks. Maybe the eslint error message should be more explicit about it, encouraging using individual properties like transitionProperty, transitionDuration and transitionTimingFunction?

**Comment by purohitdheeraj** [comment]
I can work on this , looks interesting

**Comment by IbraheemHaseeb7** [comment]
Seems fairly straightforward of an issue, can I contribute? Does it require assignee or just submit PR?

**Comment by nmn** [comment]
There's a few other missing properties in the ESLint rule. You can find them and create a PR to add them.

**Comment by xonx4l** [comment]
A Pr as per instructions of @nmn to add missing properties.

**Comment by olivierpascal** [comment]
https://github.com/facebook/stylex/blob/f968076548895dd1adc9736f1bfc371057beb94e/packages/eslint-plugin/src/stylex-valid-styles.js#L932

`lineHeight` could be a string too

**Comment by nmn** [comment]
@olivierpascal Yes, as mentioned above, we're discussing adding support for the few shorthands we manually disallowed.

**Comment by olivierpascal** [comment]
Css properties `placeContent` and `placeItems` are not supported. I guess that you are also discouraging shorthands? I am a big boy and I know what I am doing, can I use them please? ^^

**Comment by nmn** [comment]
@olivierpascal I'll add them to the list of missing properties in the ESLint plugin that should be allowed. `placeContent` and `placeItems` are not intentionally disallowed.

**Comment by necolas** [comment]
`transition` won't be allowed becuase...

> This decision was made for a few reasons:

One of the other reasons not mentioned is that we want to have consistent merging rules for properties (like React Native), and that doesn't mix with allowing shorthands that can set values for several different properties (which also has an impact on the usefulness of their Types).

**Comment by steida** [comment]
`brightness` is missing too

**Comment by nmn** [comment]
@steida There is no CSS property called brightness that I can find. It's a filter function that can be used with `filter` or `backdrop-filter`.

Can you share a link to property you think is missing?

**Comment by steida** [comment]
@nmn Ups, sorry, I typed faster than I thought.

**Comment by steida** [comment]
scrollbarWidth

**Comment by steida** [comment]
scrollSnapStop

**Comment by Daniel15** [comment]
> Any other missing property is an unknown bug so please report!

@nmn  I posted about this on Workplace too, but we should probably add `zoom`. It has some caveats but can be more useful than `transform: scale(...)` because it also affects the layout size of the element. The CSS WG recognises the value it provides and are working on standardising it, and today it works in all major browsers, except Firefox where it's gated by an `about:config` setting.

**Comment by steida** [comment]
@nmn This is perfect valid CSS.

```ts
display: "-webkit-box",
WebkitBoxOrient: "vertical",
WebkitLineClamp: 2,
```

**Comment by nmn** [comment]
Support for all the properties mentioned in this issue were added in v0.5.0 which was just released.

**Comment by steida** [comment]
@nmn lineHeight string is still not supported

**Comment by yousoumar** [comment]
@steida as far as I know, `lineHight` has been fixed in #384 and there is no release since.

**Comment by nmn** [comment]
@steida Thanks for continuing to catch these issues. Found and updated a bunch of properties in #406 just now.

if you find any additional issues, keep posting them here. I'll see them and fix them.

**Comment by nikeee** [comment]
[`fieldSizing`](https://developer.mozilla.org/en-US/docs/Web/CSS/field-sizing)

**Comment by nmn** [comment]
@nikeee This is a newer property and not something we've intentionally discouraged. Added support for it in #706

**Comment by nikeee** [comment]
[`interpolateSize`](https://developer.mozilla.org/en-US/docs/Web/CSS/interpolate-size).

If that's open for grabs, I'd file a PR.

**Comment by nmn** [comment]
@nikeee go for it.

**Comment by nikeee** [comment]
@nmn PR creation currently fails with `Pull request creation failed. Validation failed: must be a collaborator`. This is the branch:
https://github.com/facebook/stylex/compare/facebook:main...nikeee:interpolate-size

You don't have to change any contribution rules and just create the PR from that branch if you want
